source/common/network/filter_manager_impl.h

```diff
@@ -112,6 +118,34 @@ class FilterManagerImpl : protected Logger::Loggable<Logger::Id::connection> {
   void onRead();
   FilterStatus onWrite();
   bool startUpstreamSecureTransport();
+  void maybeClose();
+  void onConnectionClose(ConnectionCloseAction close_action);
+
+  void finalizeClose(ConnectionCloseAction close_action) {
+    state_.local_close_pending_ = false;
+    state_.remote_close_pending_ = false;
+    connection_.closeConnection(close_action);
+  }
```

Comment:
I think we can merge the `finalizeClose` to the maybeClose() directly and let the maybeClose() return a bool to indicate if the connection is closed. Fine methods make it's a little more to following the code.

Reply:
good idea, this part I want to make them separate to split the usage and think it is cleaner. `maybeClose` is only called by the filter callbacks, and `finalizeClose()` means filter manager has done its job, wdyt?

Reply:
done, and removed `finalizedClose` method.